### ISSUE TITLE
Automatically enable 'instant' inspection at the `hvplot` level

### ISSUE THREAD

**Issue by droumis** [holoviz/hvplot#1211]
When HoloViews supports Datashader inspections, it will still need to be enabled: at the very minimum you will need to apply the `inspect` operation appropriately.

However, at the hvPlot level, it would be appropriate to automatically configure this for you when switching to a datashaded/rasterized plot. The goal would be for datashaded plots to have all the same affordances and behaviors of normal Bokeh glyphs from the user perspective.

Technically, hvPlot should automate the generation of index layers or additional aggregate arrays so both index-based and non-index-based inspections just work.

This should include new documentation in hvPlot.

JLS note: One thing to note is that enabling instant inspection with some *random* sample (e.g `first`) could be an easy default but you probably want to have an API to allow `hvplot` to specify `where` aggregations instead (e.g. 'show the longest ship').

**Comment by maximlt** [comment]
> When HoloViews supports Datashader inspections

I'm a bit confused, HoloViews already has some support for datashader inspections? I see `inspect_points` and `inspect_polygons` in its code base at least.

> This should include new documentation in hvPlot.

As far as I can see, inspection isn't yet documented in HoloViews. I'd say it should be documented in HoloViews first, before adding documentation in hvPlot (that should link to HoloViews).

**Comment by philippjfr** [comment]
>I'm a bit confused, HoloViews already has some support for datashader inspections? I see inspect_points and inspect_polygons in its code base at least.

These are **very** crude implementations and cover a tiny fraction of what we want to support. See the [Datashader Inspections](https://github.com/orgs/holoviz/projects/9) project for the full vision.

**Comment by droumis** [comment]
Here's a more [direct link](https://github.com/orgs/holoviz/projects/9/views/3?pane=issue&itemId=46936083) to the draft issue containing the full Datashader inspections goal. 

Also, the board shows there are several things to do before implementing in hvPlot. I estimated a target of April 2024 for this.

**Comment by maximlt** [comment]
This feature should be implemented using the `selector` server-side HoverTool feature recently added to HoloViews (meant to be released in 1.21, see https://github.com/holoviz/holoviews/pull/6422). I tried it recently and think I found some usability issues, reported in https://github.com/holoviz/holoviews/issues/6565. Some more discussion around it is needed, but it might be a blocker for implementing this feature in hvPlot, which is meant to provide be a simple and user-friendly interface to HoloViews.

**Comment by maximlt** [comment]
@droumis let me know what you think please.

**Comment by droumis** [comment]
I agree that we can do a better job in HoloViews at contextualizing the hover tooltip, although I'm not sure I see this as a necessarily a blocker, especially if the renaming based on selector is automatic in HoloViews rather than requiring the user to rename.
